core/store/ledgerstore/ledger_store.go

```diff
@@ -1459,3 +1460,19 @@ func (this *LedgerStoreImp) GetContractEvent(addr common.Address) (*message.Cont
 func (this *LedgerStoreImp) GetMetaDataEvnet(height uint32, addr common.Address) (*payload.MetaDataCode, error) {
 	return this.eventStore.GetContractMetaDataEvent(height, addr)
 }
+
+func (this *LedgerStoreImp) GetShardTxState(txHash common.Uint256, notifyId uint32, isHasNotifyId bool) (*xshard_state.TxState, error) {
+	overlay := this.stateStore.NewOverlayDB()
+	xshardDB := storage.NewXShardDB(overlay)
+
+	var shardTxID xshard_types.ShardTxID
+	shardTxID = xshard_types.ShardTxID(string(txHash[:]))
+	if !isHasNotifyId {
+		return xshardDB.GetXShardState(shardTxID)
+	}
+	sink := common.NewZeroCopySink(0)
+	sink.WriteBytes([]byte(shardTxID))
+	sink.WriteUint32(notifyId)
+	crossShardTxID := xshard_types.ShardTxID(string(sink.Bytes()))
+	return xshardDB.GetXShardState(crossShardTxID)
```

Comment:
you can fetch txState from state store straightly, no need to generate xshardDB object.